src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -59,7 +60,7 @@ internal class DefaultRazorLanguageServerCustomMessageTarget : RazorLanguageServ
         FormattingOptionsProvider formattingOptionsProvider,
         IClientSettingsManager editorSettingsManager,
         LSPDocumentSynchronizer documentSynchronizer,
-        [Import(AllowDefault = true)] IOutputWindowLogger? outputWindowLogger)
+        [ImportMany] IEnumerable<ILogger> loggers)
```

Comment:
Also, we have the LoggerAdapter that already abstracts this away, and provides the convenience methods. Perhaps that just needs to move to a different level where it can be MEF exported (as an IRazorLogger it turns out 😝)?
https://github.com/dotnet/razor/blob/main/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LoggerAdapter.cs

Reply:
interestingly LoggerAdapter isn't MEF exported/imported. Maybe it should be?

Reply:
I see because of layering why it isn't. I'll just remove this and make the debugger change